test/fixtures/rules/jsx-a11y/jsx-a11y.js

```diff
@@ -0,0 +1,360 @@
+(() => {
+    // Fail
+    const bad1 = () =>
+        (<div>
+            <label>
+                Link
+            </label>
+            <input />
+        </div>);
+    const bad2 = () =>
+        (<div>
+            <label>
+                Link
+            </label>
+        </div>);
+
+    // Succeed
+    const good1 = () =>
+        (<div>
+            <label>
+                Link
+                <input />
+            </label>
+        </div>);
+    const good2 = () =>
+        (<div>
+            <label htmlFor="input">
+                Link
+            </label>
+            <input id="input" />
+        </div>);
+})();
+
+// '/lang': Enforce lang attribute has a valid value.
+(() => {
+    // Fail
+    const bad = () => <html lang="PT" />;
+
+    // Succeed
+    const good = () => <html lang="pt" />;
+})();
+
+// '/media-has-caption': Enforces that <audio> and <video> elements must have a <track> for captions.
+(() => {
+    // Fail
+    const bad1 = () => <audio />;
+    const bad2 = () => <audio> <track /></audio>;
+
+    // Succeed
+    const good = () => <audio> <track kind="captions" /></audio>;
+})();
+
+// '/mouse-events-have-key-events': Enforce that onMouseOver/onMouseOut are accompanied by onFocus/onBlur for keyboard-only users.
+(() => {
+    // Fail
+    // Disabling the following rules to showcase error
+    const bad = () => <button onMouseOut={ console.log() } />;
+
+    // Succeed
+    const good = () => <button onMouseOut={ console.log() } onBlur={ console.log() } />;
+})();
+
+// '/no-access-key': Enforce that the accessKey prop is not used on any element
+//                   to avoid complications with keyboard commands used by a screen reader.
+(() => {
+    // Fail
+    const bad = () => <div accessKey={ key } />;
+
+    // Succeed
+    const good = () => <div />;
+})();
+
+// '/no-autofocus': Enforce autoFocus prop is not used.
+(() => {
+    // Fail
+    const bad = () => <div autoFocus="true" />;
+
+    // Succeed
+    const good = () => <div />;
+})();
+
+// '/no-distracting-elements': Enforce distracting elements are not used.
+(() => {
+    // Fail
+    const bad1 = () => <marquee />;
+    const bad2 = () => <blink />;
+
+    // Succeed
+    const good = () => <div />;
+})();
+
+// '/no-distracting-elements': Enforce distracting elements are not used.
+(() => {
+    // Fail
+    const bad1 = () => <marquee />;
+    const bad2 = () => <blink />;
+
+    // Succeed
+    const good = () => <div />;
+})();
+
+// '/no-interactive-element-to-noninteractive-role': Interactive elements should not be assigned non-interactive roles.
+(() => {
+    // Fail
+    const bad = () => <input role="presentation" />;
+
+    // Succeed
+    const good = () => <input />;
+})();
+
+// '/no-noninteractive-element-interactions': Non-interactive elements should not be assigned mouse or keyboard event listeners.
+(() => {
+    // Fail
+    const bad = () => <ol onClick={ console.log() } onKeyDown={ console.log() } />;
```

Comment:
I personally don't agree with this. Let me explain why:

Given this use case:
```
<ol>
  <li /> * 10000 (10000 list items here)
</ol>
```

It's way faster and optimised to assign a onClick/onKeyDown event to the `ol` element and use the element present in the event (`event.target`).
It's not fast to assign 10000 event listeners to DOM elements.

Reply:
On the other hand (arguing with myself here 😄) this could be considered an edge-case and use a `eslint-ignore-no-noninteractive-element-interactions` inline. It looks uglier tho.

Reply:
I would also consider this an edge case, and would agree with using a ignore declaration. 

Or, at least, leave this for further consideration whenever we come across a situation like that.